pinotdb/db.py

```diff
@@ -506,11 +506,13 @@ def fetchmany(self, size=None):
     def fetchall(self):
         """
         Fetch all (remaining) rows of a query result, returning them as a
-        sequence of sequences (e.g. a list of tuples). Note that the cursor's
-        arraysize attribute can affect the performance of this operation.
+        sequence of sequences (e.g. a list of tuples).
         """
-        return list(self)
-    
+
+        results = self._results
+        self._results = None
+        return results
```

Comment:
IIUC, this change may not be safe
1.  this change the value of `self._results` to None, will affect other methods which need this result
2. the returned result is changed, previously, we change `self` to a list, now it's just part of `self` (i.e., self._results).

Can we try to fix those?

Reply:
To keep backward compatibility, maybe we can add a new method `fetchresults?` and keep the existing one unchanged?

Reply:
Thanks for taking the time to review this pull request.

It is actually an expected behaviour that a call to `fetchall` affects subsequent calls to other methods. In fact, this is already the case since returning `list(self)` will call `__iter__`, which will itself call `fetchone` until the list is emptied.

To sum up, after a call to `fetchall`, `fetchone` has been called `n` times with `n` the size of the resulting list and `self._results` is empty.

There is actually a mistake in my code, I should have set `self._results` to `[]` instead of `None`. Appart from this, the bahavior is strictly the same, except from the fact that the result is directly returned as is instead of resulting from multiple calls to `fetchone`.

Reply:
The method is already tested in :
https://github.com/python-pinot-dbapi/pinot-dbapi/blob/ef93a9039ac16db9108a9b86987bfd2d2ebb3456/tests/unit/test_db.py#L618

It needs more ?